virttest/qemu_vm.py

```diff
@@ -903,20 +903,29 @@ def add_memorys(devices, params):
             mem_params = params.object_params("mem")
             mem_params.setdefault("automem", "no")
             automem = mem_params["automem"] == "yes"
+
+            normalize_data_size = utils_misc.normalize_data_size
+            mem_size_m = "%sM" % mem_params["mem"]
+            mem_size_m = float(normalize_data_size(mem_size_m))
+            if mem_params.get("guest_max_mem"):
+                max_mem_size_m = mem_params.get("guest_max_mem")
+                max_mem_size_m = float(normalize_data_size(max_mem_size_m))
+                if mem_size_m >= max_mem_size_m:
+                    logging.debug("Guest max memory is limited to %s"
+                                  % max_mem_size_m)
+                    mem_size_m = max_mem_size_m
+                    params["mem"] = str(int(max_mem_size_m))
             if automem:
-                normalize_data_size = utils_misc.normalize_data_size
-                mem_size_m = "%sM" % mem_params["mem"]
-                mem_size_m = float(normalize_data_size(mem_size_m))
                 usable_mem_m = utils_misc.get_usable_memory_size(align=1024)
                 if mem_size_m >= usable_mem_m:
                     logging.debug("Host no enough free memory, reset guest"
                                   " memory size to %s MB" % usable_mem_m)
                     params["mem"] = str(int(usable_mem_m))
             options.append(params["mem"])
             if devices.has_device("pc-dimm"):
-                if mem_params.get("slots") and mem_params.get("maxmem"):
+                if mem_params.get("slots") and mem_params.get("maxmem_mem"):
```

Comment:
This change isn't correct. `maxmem_mem` was renamed to `maxmem` by `mem_params = params.object_params("mem")`.